runtime/sema/check_expression.go

```diff
@@ -264,23 +297,7 @@ func (checker *Checker) visitValueIndexingExpression(
 	indexingExpression ast.Expression,
 	isAssignment bool,
 ) Type {
-	indexingType := indexingExpression.Accept(checker).(Type)
-
-	elementType := indexedType.ElementType(isAssignment)
-
-	// check indexing expression's type can be used to index
-	// into indexed expression's type
-
-	if !indexingType.IsInvalidType() &&
-		!IsSubType(indexingType, indexedType.IndexingType()) {
+	checker.VisitExpression(indexingExpression, indexedType.IndexingType())
 
-		checker.report(
-			&NotIndexingTypeError{
-				Type:  indexingType,
-				Range: ast.NewRangeFromPositioned(indexingExpression),
-			},
-		)
-	}
```

Comment:
I guess `NotIndexingTypeError` isn't really useful, as a type mismatch error is also reported and much more useful to fix the problem